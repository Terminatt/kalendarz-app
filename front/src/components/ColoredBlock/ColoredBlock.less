@import '~styles/variables.less';

.colored-block {
    color: @black;
    text-align: center;
    width: 100px;
    padding: @padding-minimal @padding-small 2px @padding-small;
}