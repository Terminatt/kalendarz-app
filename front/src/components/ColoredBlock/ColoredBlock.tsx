import { GenericReactContent } from '@generics/generics';
import { joinClassNames } from '@utils/general';
import React from 'react';

import './ColoredBlock.less';

export interface ColoredBlock {
    bgColor: string;
    className?: string;
    children?: GenericReactContent;
    style?: React.CSSProperties
}

// TODO Calculate text color based on room type background color
const ColoredBlock: React.FC<ColoredBlock> = (props) => {
    const {
        bgColor, children, className, style,
    } = props;
    return (
        <div style={{ backgroundColor: bgColor, ...style }} className={joinClassNames(['colored-block', className])}>
            {children}
        </div>
    );
};

export default ColoredBlock;
