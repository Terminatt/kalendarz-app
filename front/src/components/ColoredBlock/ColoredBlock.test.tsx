import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import ColoredBlock from './ColoredBlock';

describe('ColoredBlock Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer
            .create(
                <ColoredBlock bgColor="#019267">
                    Test
                </ColoredBlock>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('renders proper color', () => {
        const container = document.createElement('div');
        const color = "#019267"

        render(
            <ColoredBlock bgColor={color}>
                Test
            </ColoredBlock>, container
        )
        
        expect(container.firstChild).toHaveStyle(`background-color: ${color}`);
    });

    it('has correct content', () => {
        const container = document.createElement('div');
        const color = "#019267"
        const text = 'Test'

        render(
            <ColoredBlock bgColor={color}>
                {text}
            </ColoredBlock>, container
        )
        
        expect(container.firstChild?.textContent).toBe(text);
    });
});
