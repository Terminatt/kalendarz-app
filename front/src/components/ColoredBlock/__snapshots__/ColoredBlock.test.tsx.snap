// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColoredBlock Component matches the snapshot 1`] = `
<div
  className="colored-block"
  style={
    Object {
      "backgroundColor": "#019267",
    }
  }
>
  Test
</div>
`;
